src/utils.c

```diff
@@ -182,6 +182,131 @@ ts_integer_to_internal(Datum time_val, Oid type_oid)
 	}
 }
 
+int64
+ts_time_value_to_internal_or_infinite(Datum time_val, Oid type_oid,
+									  TimevalInfinity *is_infinite_out)
+{
+	switch (type_oid)
+	{
+		case TIMESTAMPOID:
```

Comment:
i think you should merge TIMESTAMPOID and TIMESTAMPTZOID branch since the code is identical

Reply:
not quite identical, they're still different types